packages/universal/verify/src/verify.ts

```diff
@@ -40,33 +47,20 @@ export function verify(
   }
 }
 
-verify.noop = (() => {
-  /** noop */
-}) as unknown as Exclude<typeof verify, "noop">;
-
-function noop(): void {
-  return;
-}
-
-verify.noop = noop as unknown as VerifyFn;
-
 export function verified<T, U extends T>(
   value: T,
   check: (input: T) => input is U,
   error?: Expectation<T>,
-): U {
-  verify(value, check, error);
-  return value;
-}
-
-verified.noop = <const T, const U extends T>(
-  value: T,
-  _check: (input: T) => input is U,
-  _error?: Expectation<T>,
-): U => {
-  return value as U;
-};
+): U;
 
+export function verified(v: unknown): unknown {
+  if (import.meta.env.DEV) {
+    // eslint-disable-next-line prefer-rest-params
+    const [value, check, error] = [...arguments] as Parameters<typeof verified>;
```

Comment:
this hides extra function params from terser, so terser can optimize the prod build and inline this function.
terser will then replave `verified(x)` with `x`